app/importers/mahonia_mapper.rb

```diff
@@ -28,9 +28,27 @@ def fields
     end
   end
 
+  def creator_name(first_name: first, last_name: last)
+    "#{first_name} #{last_name}".strip
+  end
+
+  # return first and last names of 1-n authors
+
   def creator
-    return if metadata['author1_fname'].nil? && metadata['author1_lname'].nil?
-    Array("#{metadata['author1_fname']} #{metadata['author1_lname']}")
+    creators = []
+
+    number_of_authors = metadata.keys.select do |k|
+      k.match?(/^author[0-9]*_lname$/)
+    end.count
+
+    unless metadata["author1_fname"].nil? && metadata["author1_lname"].nil?
+
+      (1..number_of_authors).each do |i|
+        creators << creator_name(first_name: metadata["author#{i}_fname"], last_name: metadata["author#{i}_lname"])
+      end
+    end
+
+    creators unless creators.empty?
```

Comment:
What do we return if `creators` is empty? It looks like it's the output of `unless` which may not be what we want. In either case, I think an explicit return value will improve readability. What do you think of?

```ruby
...
return nil if creators.empty? # is `nil` correct, or `[]`?
creators
```

Reply:
Yes, that is more readable, ++. I want nil to be returned, not an empty array.